src/device_info.sh

```diff
@@ -0,0 +1,173 @@
+# This file deals with functions related to hardware information
+
+include "$KW_LIB_DIR/remote.sh"
+include "$KW_LIB_DIR/kwlib.sh"
+
+declare -gA shared_data=(['ram']='' ['cpu_model']='' ['cpu_frequency']='' ['disks']='' ['os']='')
+
+# This function populates the ram element from the shared_data global variable
+# with the total RAM memory from the target machine in kB
+#
+# @target Target machine
+function get_ram()
+{
+  local target="$1"
+  local ram
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      cmd="cat /proc/meminfo | grep 'MemTotal' | grep -o '[0-9]*'"
+      ram=$(cmd_manager "SILENT" "$cmd")
+      ;;
+    3) # REMOTE
+      echo TODO
+      ;;
+  esac
+  shared_data['ram']="$ram"
+}
+
+# This function provides the model and frequency of the CPU from a machine
+#
+# @target Target machine
+function get_cpu()
+{
+  local target="$1"
+  local cpu_model
+  local cpu_frequency
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      cmd="lscpu | grep 'Model name:' | sed -r 's/Model name:\s+//g'"
+      cpu_model=$(cmd_manager 'SILENT' "$cmd")
+      cmd="lscpu | grep MHz | sed -r 's/(CPU.*)/\t\t\1/'"
+      cpu_frequency=$(cmd_manager 'SILENT' "$cmd")
+      ;;
+    3) # REMOTE
+      echo TODO
+      ;;
+  esac
+
+  shared_data['cpu_model']="$cpu_model"
+  shared_data['cpu_frequency']="$cpu_frequency"
+}
+
+function get_disks()
+{
+  local target="$1"
+  local disks
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      cmd="lsblk -d -o name,rota | grep -v loop | tail -n +2 |"
+      cmd+="sed -r 's/(.*)\s+1/\t\1rotational/;s/(.*)\s+0/\t\1not rotational/'"
+      disks=$(cmd_manager "SILENT" "$cmd")
+      ;;
+    3) # REMOTE
+      echo TODO
+      ;;
+  esac
+
+  shared_data['disks']="$disks"
+}
+
+function get_os()
+{
+  local target="$1"
+  local os
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      os=$(hostnamectl | grep 'Operating System' | sed -r 's/.*:\s(.*)/\1/')
```

Comment:
That's a nice approach, but we can get the OS information without external tools. Actually, we already have it :)
See `which_distro` and `detect_distro`, maybe we can move some of this function to your new file.